rfcs/text/0000-bundle-definition.md

```diff
@@ -0,0 +1,248 @@
+
+## Detailed Design
+
+### What is a bundle?
+
+A bundle is a unit of data organization, roughly equivalent to a folder or directory. In the HCA DCP, a bundle is
+defined as either a transactional grouping of files that belong together for scientific reasons, or a useful association
+between a set of files for utility purposes.
+
+* Data bundles contain the results of a single data transaction. They can be viewed as the digital equivalent of an
+  "instrument run" - a quantity of data gathered from a single process instance. Data bundles are annotated with
+  metadata using a procedure designed by the metadata team.
+
+* Utility bundles exist to organize (group) data bundles and represent data of non-DCP origin (such as reference data)
+  that is used in the course of analysis.
+
+### How are bundle types defined?
+
+Operationally, bundle types can be defined like file types or media types: they are used as a data type identifier for
+applications to pattern match on whether data should be considered as suitable for input to the application.
+
+### Bundle types and their definitions
+
+Bundle types, definitions, consumers, and specifications are outlined below. Use cases from above are indicated where
+supported by a specific bundle type.
+
+#### Table header definitions
+
+- **Type**: The name of the bundle type
+
+- **Definition**: What is and is not in this bundle type
+
+- **Target consumer**: Who will be consuming this bundle type
+
+- **Use cases**: Which Users Stories are supported by this bundle type
+
+| Type | Definition | Target consumer | Use cases |
+|:-|:-|:-|:-|
+| Primary bundle | A bundle that contains primary (raw) data files and all related metadata files. | computational biologists; Data Processing Pipelines; Data Browser; Query Service | 1 |
+| Secondary bundle | A bundle that contains secondary (alignment, expression) data files produced by the Data Processing Pipelines, input primary data files, and all related metadata files. | computational biologists; Data Processing Pipelines; Data Browser; Query Service; Matrix Service | 2, 3, 4(?) |
+| Tertiary bundle* | A bundle that contains tertiary (expression) data files produced by the Matrix Service, input primary and secondary data files, and all related metadata files. | computational biologists; biologists; Data Browser; Query Service | 3, 4 |
+| Project bundle* | A bundle that contains project-level metadata files. This bundle type does not contain data files or metadata files related to biomaterials, protocols, processes, or files. | computational biologists; biologists; | ? |
+| Resource/Reference bundle* | A bundle that contains reference data files used by the Data Processing Pipelines(?) and all related metadata files. This bundle type does not contain data files or metadata files related to biomaterials, protocols, processes(, or projects?). | computational biologists; Data Processing Pipelines; | 2, 3, 5 |
```

Comment:
Is it possible to give a bit more detail on the resource bundle? What kind of data goes here? What is the current process for a user to obtain this kind of data?

Reply:
This was a good gap to address. I added to the *New Resource/Reference bundle* section below:
```
#### New Resource/Reference bundle

+This is a type of bundle that supports reference genomes and other accessory inputs to analysis pipelines. They should
+be represented in the DCP Data Store to achieve completeness and durability of analysis inputs, which in turn is
+necessary for analysis reproducibility.

+- The Data Processing Pipelines team is responsible for generating these.
```

Reply:
Currently the user would need to look at analysis inputs to recover the URLs for these inputs, and hope that they are still there.